### ISSUE TITLE
Use of ONIX BIC subject categories

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#9906]
There was a wish/requirement to provide autosuggestion of BIC subject categories codes in OMP 3.3. and to use them in ONIX export.
The ONIX export expects the BIC subject categories, s. https://github.com/pkp/omp/blob/stable-3_3_0/plugins/importexport/onix30/filter/MonographONIX30XmlFilter.inc.php#L408-L425.
However, we do not use them for the subject field, s. https://github.com/pkp/pkp-lib/blob/stable-3_3_0/classes/components/forms/publication/PKPMetadataForm.inc.php#L59-L68 and https://github.com/pkp/pkp-lib/blob/stable-3_3_0/api/v1/vocabs/PKPVocabHandler.inc.php#L89-L92.
(Differently to the filed languages, that provides the ONIX list of languages, s. https://github.com/pkp/omp/blob/stable-3_3_0/api/v1/vocabs/VocabHandler.inc.php#L53-L57).

We have the codes here: https://github.com/pkp/pkp-lib/blob/stable-3_3_0/locale/en_US/bic21subjects.xml.
Also we have classes that deal with code lists, s. https://github.com/pkp/omp/tree/stable-3_3_0/classes/codelist.
There is also a class Subject and SubjectDAO, s. https://github.com/pkp/omp/blob/stable-3_3_0/classes/codelist/Subject.inc.php and https://github.com/pkp/omp/blob/stable-3_3_0/classes/codelist/SubjectDAO.inc.php.
But it seems they were never used.
(I have not tried to integrate it all yet, so that the BIC codes can be used in subject fields).

Furthermore, BIC codes are obsolete now, s. https://ns.editeur.org/bic_categories, https://bic.org.uk/resources/bic-standard-subject-categories/ and https://bic.org.uk/resources/bic-standard-subject-categories/obsolescence-of-bic-codes-and-transition-to-themabics-operational-faqs/.
According to the FAQs: if BIC codes are still considered when provided in ONIX, but the recommendation is to use Thema (https://bic.org.uk/resources/thema/).

Thus, maybe it is a question if we would like to use BIC codes because we already have some structure for it -- it seems that implementation would not take so much time -- or if we would like to immediately change to Thema -- which will probably mean some more changes and would take longer.

Thus, either we

**Comment by kaitlinnewson** [comment]
Since BIC is obsolete as of February 2024 (and [DOAB has moved away from it as well](https://www.doabooks.org/blog/?link=https%3A%2F%2Foapen.hypotheses.org%2F976)), it would make more sense to implement Thema subject categories in the autosuggestion and Onix export.

**Comment by asmecher** [comment]
Agreed, @kaitlinnewson -- but @bozana, it looks like your initial comment was cut off; can you review?

**Comment by bozana** [comment]
Hmmm... I do not know any more what I wanted to write there, and why is it not there :smile: 
Else, yes, I agree that we should rather implement Thema...

**Comment by kaitlinnewson** [comment]
Hi @bozana, the PRs above are ready for your review when you have some time - thanks!
